plugins/status/plugin.go

```diff
@@ -480,16 +457,17 @@ func (p *Plugin) reconfigure(config interface{}) {
 	p.logger.Info("Status reporter configuration changed.")
 
 	if newConfig.Prometheus && !p.config.Prometheus {
-		p.registerAll()
+		p.collectors.RegisterAll(p.manager.PrometheusRegister(), p.logger)
 	} else if !newConfig.Prometheus && p.config.Prometheus {
-		p.unregisterAll()
+		p.collectors.UnregisterAll(p.manager.PrometheusRegister())
+	} else if newConfig.Prometheus && p.config.Prometheus {
+		p.collectors.ReregisterCustomizedCollectors(p.manager.PrometheusRegister(), newConfig.PrometheusConfig, p.logger)
```

Comment:
We could add a check here to see if the config has actually changed.

Reply:
think again, actually this is already checked at line 452 https://github.com/open-policy-agent/opa/pull/7156/files#diff-1bcf4e3a0c44d66ff6e1478c913e89580f85f9b0ff0e3274148db21bbef9e085R452